src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java

```diff
@@ -121,6 +144,204 @@ public void importRepositories(final ProgrammingExercise templateExercise, final
+                copyExerciseContentToRepositoryForHaskellExercise(solutionRepo, RepositoryType.SOLUTION, basePath);
+                copyExerciseContentToRepositoryForHaskellExercise(testRepo, RepositoryType.TESTS, basePath);
+            }
+        }
+    }
+
+    private void copyExerciseContentToRepositoryForExerciseWithPlainDirectoryStructure(Repository repository, RepositoryType repositoryType, Path exerciseDir) throws IOException {
+        for (Path file : retrieveRepositoryContentPaths(retrieveRepositoryDirectoryPath(exerciseDir, repositoryType.getName()))) {
+            repositoryService.createFile(repository, String.valueOf(file.getFileName()), Files.newInputStream(file));
+
+        }
+
+    }
+
+    private void copyExerciseContentToRepositoryForPythonExercise(Repository repository, RepositoryType repositoryType, Path exerciseDir) throws IOException {
+        boolean testRepo;
+        if (repositoryType == RepositoryType.TESTS) {
+            repositoryService.createFolderRecursively(repository, "behavior");
+            repositoryService.createFolderRecursively(repository, "structural");
+            testRepo = true;
+        }
+        else {
+            testRepo = false;
+        }
+
+        for (Path file : retrieveRepositoryContentPathsPython(exerciseDir, repositoryType)) {
+            if (testRepo) {
+                if (file.toAbsolutePath().toString().contains("behavior")) {
+                    repositoryService.createFile(repository, "behavior/" + file.getFileName(), Files.newInputStream(file));
+                }
+                else {
+                    repositoryService.createFile(repository, "structural/" + file.getFileName(), Files.newInputStream(file));
+                }
+            }
+            else {
+
+                repositoryService.createFile(repository, String.valueOf(file.getFileName()), Files.newInputStream(file));
+            }
+
+        }
+    }
+
+    private void copyExerciseContentToRepositoryForOcamlExercise(Repository repository, RepositoryType repositoryType, Path exerciseDir) throws IOException {
+        boolean testRepo;
+        if (repositoryType == RepositoryType.TESTS) {
+            repositoryService.createFolderRecursively(repository, "checker");
+            repositoryService.createFolderRecursively(repository, "overrides");
+            repositoryService.createFolderRecursively(repository, "solution");
+            repositoryService.createFolderRecursively(repository, "test");
+            repositoryService.createFolderRecursively(repository, "assignment");
+            testRepo = true;
+        }
+        else {
+            testRepo = false;
+            repositoryService.createFolderRecursively(repository, "src");
+        }
+
+        for (Path file : retrieveRepositoryContentPathsOcaml(exerciseDir, repositoryType)) {
+            if (testRepo) {
+                if (file.toAbsolutePath().toString().contains("checker")) {
```

Comment:
Possible issue similar to `behavior` above.

Reply:
no longer part of the implementation